libraries/AP_RangeFinder/RangeFinder_Backend.h

```diff
@@ -80,8 +79,12 @@ class AP_RangeFinder_Backend
     void set_status(RangeFinder::RangeFinder_Status status);
 
     RangeFinder::RangeFinder_State &state;
-    MAV_DISTANCE_SENSOR sensor_type;
 
     // semaphore for access to shared frontend data
     AP_HAL::Semaphore *_sem;    
+
+    virtual MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const {
+        // most of the backends are lasers:
+        return MAV_DISTANCE_SENSOR_LASER;
```

Comment:
Interesting.  Not sure I agree with your reasoning entirely.  But making it pure-virtual OTOH was (making the driver declare) was something I considered and ruled out in that I thought it would increase text size too much.  So now I've measured:

```
without pure-virtual: bin/arducopter  974664  2864  55608  1033136
with pure-virtual:    bin/arducopter  974680  2864  55608  1033152
```

I've updated this patch to use the pure-virtual approach...